src/transformState.js

```diff
@@ -5,7 +5,26 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      default:
+        for (const key of action.keysToRemove) {
+          if (state[key]) {
+            delete state[key];
+          }
+        }
```

Comment:
This is not a default case, it's removeProperties case. Default case should handle situation when action.type is neither clear, addProperties nor removeProperties, by throwing an error "This action type is not supported", for example